test/unit/storage_async_test.rb

```diff
@@ -302,10 +338,42 @@ def assert_sentinel_config(conf, conn)
       host, port = endpoint.address
       assert_equal conf[:sentinels][i][:host], host
       assert_equal conf[:sentinels][i][:port], port
-      assert_equal(conf[:sentinels][i][:password], password) if conf[:sentinels][i].key? :password
     end unless conf[:sentinels].empty?
   end
 
+  def assert_acl_credentials(conf, client)
+    unless conf[:username].to_s.strip.empty? && conf[:password].to_s.strip.empty?
+      assert_instance_of ThreeScale::Backend::AsyncRedis::Protocol::ExtendedRESP2, client.protocol
+      username, password = client.protocol.instance_variable_get(:@credentials)
+      assert_equal conf[:username], username
+      assert_equal conf[:password], password
+    end
+
+    unless conf[:sentinel_username].to_s.strip.empty? && conf[:sentinel_password].to_s.strip.empty?
+      sentinel_username, sentinel_password = client.instance_variable_get(:@sentinel_credentials)
+      assert_equal conf[:sentinel_username], sentinel_username
+      assert_equal conf[:sentinel_password], sentinel_password
+    end
```

Comment:
should we test that `sentinel_username` is `nil` when `conf[:sentinel_username].to_s.strip.empty?`? Now we only do positive testing as far as I can tell.

Reply:
Done in https://github.com/3scale/apisonator/pull/391/commits/fef9bcd273ebab868841b3900c811ea9a9ba5699